ui/src/components/TaskRunsCard/index.jsx

```diff
@@ -270,16 +274,20 @@ export default class TaskRunsCard extends Component {
                   )}
                   <Typography>{artifact.name}</Typography>
                 </div>
-              </TableCell>
-              <TableCell
-                className={classNames(
-                  classes.linkCell,
-                  classes.artifactsTableCell
-                )}>
+              </Link>
+            </TableCell>
+            <TableCell
+              className={classNames(
+                classes.linkCell,
+                classes.artifactsTableCell
+              )}>
+              <Link
+                className={classes.artifactLink}
+                to={this.getArtifactUrl(artifact)}>
                 {artifact.isPublic && <LinkIcon />}
                 {!artifact.isPublic && artifact.url && <OpenInNewIcon />}
-              </TableCell>
-            </Link>
+              </Link>
+            </TableCell>
```

Comment:
That's a lot better! The only problem I see however is that paddings are not clickable so we might need to do something else. I was thinking we can put everything in one table cell and style the cell using flexbox.

```diff
diff --git a/ui/src/components/TaskRunsCard/index.jsx b/ui/src/components/TaskRunsCard/index.jsx
index 172dffaa1..2bf3a1bf3 100644
--- a/ui/src/components/TaskRunsCard/index.jsx
+++ b/ui/src/components/TaskRunsCard/index.jsx
@@ -65,9 +65,6 @@ const DOTS_VARIANT_LIMIT = 5;
     pointer: {
       cursor: 'pointer',
     },
-    linkCell: {
-      textAlign: 'right',
-    },
     logButton: {
       marginRight: theme.spacing.unit,
     },
@@ -86,22 +83,16 @@ const DOTS_VARIANT_LIMIT = 5;
     },
     artifactLink: {
       textDecoration: 'none',
-      width: '100%',
-      display: 'block',
-      height: '100%',
+      display: 'flex',
+      justifyContent: 'space-between',
       verticalAlign: 'middle',
     },
     artifactTableRow: {
       height: 'auto',
     },
-    artifactNameTableCell: {
-      width: '100%',
-    },
-    artifactsTableCell: {
-      padding: `${theme.spacing.unit}px ${theme.spacing.triple}px`,
-    },
     artifactNameWrapper: {
       display: 'inline-flex',
+      flexBasis: '50%',
     },
     liveLogLabel: {
       marginLeft: theme.spacing.unit / 2,
@@ -240,22 +231,14 @@ export default class TaskRunsCard extends Component {
               }
             )}
             hover={!!artifact.url}>
-            <TableCell className={classes.artifactsTableCell}>
-              <Link
-                className={classes.artifactLink}
-                to={this.getArtifactUrl(artifact)}>
-                {artifact.isPublic && <LockOpenOutlineIcon />}
-                {!artifact.isPublic && artifact.url && <LockIcon />}
-              </Link>
-            </TableCell>
-            <TableCell
-              className={classNames(
-                classes.artifactNameTableCell,
-                classes.artifactsTableCell
-              )}>
+            <TableCell>
               <Link
                 className={classes.artifactLink}
                 to={this.getArtifactUrl(artifact)}>
+                <div>
+                  {artifact.isPublic && <LockOpenOutlineIcon />}
+                  {!artifact.isPublic && artifact.url && <LockIcon />}
+                </div>
                 <div className={classes.artifactNameWrapper}>
                   {artifact.isLog && (
                     <Label status="info" mini className={classes.logButton}>
@@ -264,18 +247,10 @@ export default class TaskRunsCard extends Component {
                   )}
                   <Typography>{artifact.name}</Typography>
                 </div>
-              </Link>
-            </TableCell>
-            <TableCell
-              className={classNames(
-                classes.linkCell,
-                classes.artifactsTableCell
-              )}>
-              <Link
-                className={classes.artifactLink}
-                to={this.getArtifactUrl(artifact)}>
-                {artifact.isPublic && <LinkIcon />}
-                {!artifact.isPublic && artifact.url && <OpenInNewIcon />}
+                <div>
+                  {artifact.isPublic && <LinkIcon />}
+                  {!artifact.isPublic && artifact.url && <OpenInNewIcon />}
+                </div>
               </Link>
             </TableCell>
           </TableRow>
```

Let me know what you think.

Reply:
I think this is a much better way to fix this issue.
@helfi92